src/rgw/rgw_acl.h

```diff
@@ -278,6 +254,26 @@ struct ACLReferer {
     DECODE_FINISH(bl);
   }
   void dump(Formatter *f) const;
+
+private:
+  boost::optional<boost::string_ref> get_http_host(const boost::string_ref url) const {
+    size_t pos = url.find("://");
+    if (pos == boost::string_ref::npos || url.starts_with("://") ||
+        url.ends_with("://") || url.ends_with('@')) {
+      return boost::string_ref(nullptr, 0);
```

Comment:
I would propose returning `boost::none` (according to one of the comments above).